docs/how-tos/production_configuration.rst

```diff
@@ -0,0 +1,245 @@
+.. _production_configuration:
+
+Configure Aspects for Production
+********************************
+
+Choosing an xAPI Pipeline
+#########################
+
+Aspects can be configured to send xAPI events to ClickHouse in several different ways. Which one you choose depends on your specific organization's needs, deployment infrastructure, scale, and familiarity with different technology. The `event-routing-backends docs`_ have more details on the configuration options outline here.
+
+At a high level the options are:
+
+**Celery tasks without batching (default)**
+
+Each tracking log event that has an xAPI transform gets queued as a task. The task performs the xAPI transform and queues a second task to send the event to Ralph. Ralph checks for the existence of the event in ClickHouse before inserting. Events deemed "`business critical`_" can be configured to be retried upon failure.
+
+Pros:
+
+- Celery is a familiar technology for Open edX site operators and can be configured to run in a highly available and scalable way with multiple workers and things like AWS SQS as a backend
+- Uses exiting systems and processes, so has the least configuration
+- This is the best tested path and least likely to have surprising bugs in v1
+
+Cons:
+
+- The duplication of tasks combined with the high volume of tasks and latency of making network requests for each event can quickly overwhelm Celery workers, requiring many workers to be run and causing delays for other Celery tasks
+- Other processes that spawn many or slow Celery tasks, such as re-grading a large course, or many successive course publish events for large courses, can delay event delivery
+- Downstream outages in Ralph of ClickHouse can exacerbate these issues with many pending retries piling up
+- ClickHouse is much less efficient with many small inserts, resulting in the possibility of insert delays or even errors if there are many simultaneous single row inserts
+
+Recommended for:
+
+Development, testing, or very small deployments.
+
+
+**Celery tasks with batching**
+
+Event-routing-backends can be configured to `batch requests`_ to Ralph, mitigating many of the issues above while still keeping the simplicity of configuration. Batching is accomplished by a "this many or this long" check, so even on low traffic deployments events will only be delayed by a fixed amount of time. In load testing, batching at up to 1000 events allowed for loads over 50 events/sec on a single worker, which is enough for most production instances.
+
+Pros:
+
+- The same as Celery above
+- Far fewer Celery tasks, reducing issues around worker contention and task delays
```

Comment:
Task times are improved too, from 0.3 seconds to 0.005 ~ 0.01 (it depends on Redis latency)